### PR TITLE
Feature/rtl audit moving lint module to bintray

### PR DESCRIPTION
**Pull request by khaykov**
I moved the [WordPressLint library](https://github.com/wordpress-mobile/WordPress-Lint-Android) to the bintray, so this PR adds it as a remote dependency to submodules and as a project dependency to the main module (as a subtree).

@aforcier thanks for the tip! Maybe you can also take a look to make sure everything is ok?

### CODE REVIEW

**Comment by aforcier** [comment]
Nice @khaykov! Will give this a more thorough pass through soon, but in general I'm wondering, how often would you expect we need to make changes to the lint library? Do you think we could possibly drop the subtree entirely and use the bintray artifact in the base project too, or would that add too much overhead?

**Comment by khaykov** [comment]
@aforcier I really do hope that we will make a lot of changes to the library, but in reality, it will probably be relatively rare. The thing is - having it as a subtree is really convenient for testing new issues on a real android project. Maybe to get around it one can temporarily add it as a subtree, add/test new issues, push changes back to subtree it and remove it from the project (or delete a branch altogether)? What do you think?

**Comment by aforcier** [comment]
@khaykov I think that could work - probably in most cases we could get away with just using maven local, following the steps in the README to ` ./gradlew assemble test publishToMavenLocal` and then importing that build in the WordPress project temporarily. Then we'd push changes directly to the repo, without a subtree. Probably a matter of preference.

File: libs/lint/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+* In your build.gradle:
+```groovy
+dependencies {
+    compile 'org.wordpress:lint:1.0.0' // use version 1.0.0
+}
+```
+
+## Publish an updated version to your local maven repository
+
+You can bump the [version name in the main build file: `WordPressLint/build.gradle`][1]. After updating the build file, you can build, and publish the library to your local maven repo. That will let you try the new version in your app for example.
```
Comment (aforcier) [inline libs/lint/README.md]: The reference [1] is missing here, so this link doesn't work.

File: libs/lint/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+When a new version is ready to be published to the remote repository, use the following command to upload it to Bintray:
+
+```shell
+$ ./gradlew assemble test bintrayUpload -PbintrayUser=FIXME -PbintrayKey=FIXME -PdryRun=false
+```
+
+## License ##
+
+WordPress-Login-Flow-Android is an Open Source project covered by the
```
Comment (aforcier) [inline libs/lint/README.md]: Copy/paste strikes again 😉

**Comment by aforcier** [comment]
Did a test `subtree push` and build for the login lib to try things out from this branch, all looks good now 👍.

**Comment by khaykov** [comment]
@aforcier Thanks! Yes, I feel like this approach makes the most sense given the size of the lib. We can always import it as a subtree at some point, if necessary.
I removed the submodule, and made sure that custom rules are still working in main module 👍

**Comment by aforcier** [comment]
@khaykov great! :shipit: from me (those README issues obviously can't be addressed from this end anymore 🙃), down to you @malinajirka!

**Comment by khaykov** [comment]
Good catch! I fixed the readme issues in the library 🍡

**Comment by malinajirka** [comment]
Thanks for this PR @khaykov! I've tested it and it looks good;). 

I agree that having it as a dependency seems like the most reasonable solution!

**Pull request merged**